src/main/java/relay/data_access/FirebaseCourseDataAccessObject.java

```diff
@@ -0,0 +1,114 @@
+
+package relay.data_access;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+
+import relay.entity.Course;
+import relay.entity.Instructor;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+
+public class FirebaseCourseDataAccessObject {
+
+
+    /**
+     * Creates a new course document in Firestore based on the given Course object.
+     *
+     * @param course The Course object containing course details to be added to Firestore.
+     */
+    public void createCourse(Course course) {
+        Map<String, Object> courseDocument = new HashMap<>();
+        courseDocument.put("courseID", course.getCourseID());
+        courseDocument.put("courseName", course.getCourseName());
+        courseDocument.put("instructorID", course.getInstructor().getInstructorID());
+
+        ApiFuture<WriteResult> future = FirestoreSingleton.get().collection("courses").document(course.getCourseID()).set(courseDocument);
+        try {
+            System.out.println("Update time : " + future.get().getUpdateTime());
+        } catch (InterruptedException | ExecutionException e) {
+            System.out.println(e);
+        }
+
+    }
+
+    /**
+     * Retrieves a list of courses associated with a specific instructor from the Firestore database.
+     *
+     * @param instructor The Instructor object for which courses are to be retrieved.
+     * @return An ArrayList of Course objects associated with the specified instructor.
+     * @throws ExecutionException   If the execution of the query fails.
+     * @throws InterruptedException If the query execution is interrupted.
+     */
+    public ArrayList<Course> getCoursesByInstructor(Instructor instructor) throws ExecutionException, InterruptedException {
```

Comment:
Can we remove this throw or does Java give erorr?

Reply:
If it gvies error, just convert to try-catch

Reply:
Gave error, so placed for loop in try-catch